enhancements/ingress/set-replace-append-delete-deny-headers.md

```diff
@@ -0,0 +1,956 @@
+                  value: /my-first-blog-post
+            - name: X-SSL-Client-Cert
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,base64]"
+            - name:   Content-Language
+              action:
+                type: Delete
+            - name: X-Target
+              action:
+                type: Set
+                set:
+                  value: "%[req.hdr(host),lower]"
+```
+2. Route owner wants to set/delete HTTP response or request headers for only a specific route.
+   Users edits/creates the Route having Spec as follows -
+```yaml
+      apiVersion: route.openshift.io/v1
+      kind: Route
+      metadata:
+        labels:
+          type: sharded
+        name: hello-openshift-actions
+        namespace: hello-openshift
+      spec:
+        subdomain: hello-openshift
+        tls:
+          termination: edge
+        to:
+          kind: Service
+          name: hello-openshift
+        httpHeaders:
+          actions:
+            response:
+            - name: X-Frame-Options
+              action:
+                type: Set
+                set:
+                  value: DENY
+            - name: X-Cache-Info
+              action:
+                type: Set
+                set:
+                  value: "not cacheable; meta data too large"
+            - name: X-XSS-Protection
+              action:
+                type: Delete
+            - name: X-Source
+              action:
+                type: Set
+                set:
+                  value: "%[res.hdr(X-Value),lower]"
+            request:
+            - name: Content-Location
+              action:
+                type: Set
+                set:
+                  value: /my-first-blog-post
+            - name: X-SSL-Client-Cert
+              action:
+                type: Set
+                set:
+                  value: "%{+Q}[ssl_c_der,base64]"
+            - name:   Content-Language
+              action:
+                type: Delete
+            - name: X-Target
+              action:
+                type: Set
+                set:
+                  value: "%[req.hdr(host),lower]"
+```
+
+### API Extensions
+
+A)  Setting or deleting HTTP request or response headers via `IngressController.spec`.
+The existing `IngressControllerHTTPHeaders` Go type definition of the IngressController API is extended by adding a field for header manipulations.
+```go
+// IngressControllerSpec is the specification of the desired behavior of the
+// IngressController.
+type IngressControllerSpec struct {
+    // httpHeaders defines policy for HTTP headers.
+    //
+    // If this field is empty, the default values are used.
+    //
+    // +optional
+    HTTPHeaders *IngressControllerHTTPHeaders `json:"httpHeaders,omitempty"`
+}
+```
+
+```go
+// IngressControllerHTTPHeaders specifies how the IngressController handles
+// certain HTTP headers.
+type IngressControllerHTTPHeaders struct {
```

Comment:
Can you update this from openshift/api?  it is called `IngressControllerHTTPHeaderActions` there.